snowexsql/api.py

```diff
@@ -404,16 +399,17 @@ def from_area(
                 where_clauses = []
                 params = {}
                 
-                # Add spatial filter - compare in SRID 4326 since that's
-                # what data is stored in
+                # Add spatial filter - transform geometries to same CRS
+                # Transform DB geometry to match the search CRS
                 if needs_site_join:
                     where_clauses.append(
-                        f"ST_Intersects(s.geom, ({search_geom_sql}))"
+                        f"ST_Intersects(ST_Transform(s.geom, {crs}), "
+                        f"({search_geom_sql}))"
```

Comment:
We should keep the approach of using the SRID that is stored in the DB geom column. Otherwise the query can not use the search index and will scan the entire table, which will make this very slow